### PR TITLE
2528 Replace Incremental Index Global Flags with Getters

### PR DESCRIPTION
**Pull request by justinborromeo**
Some parameters in the addToFacts function of IncrementalIndex are replaced with getters for their corresponding class member variables.  This change addresses #2528 which is based on the discussion in #2519.  

The parameters removed were:
Boolean: deserializeComplexMetrics
Boolean: reportParseExceptions
AtomicInteger: numEntries
AggregatorFactory[]: metrics
AtomicLong[]: sizeInBytes/bytesInMemory

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java [hunk 1]
```diff
@@ -785,9 +780,29 @@ public int size()
...
+  {
+    return reportParseExceptions;
+  }
+
+  public AtomicInteger getNumEntries()
+  {
+    return numEntries;
+  }
+
+  public AggregatorFactory[] getMetrics()
```
Comment (surekhasaharan) [inline processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java]: Do all of these getters need to be public ? I think except `getBytesInMemory()`, others can be default package access.

File: processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java [hunk 2]
```diff
@@ -156,7 +153,9 @@ protected AddToFactsResult addToFacts(
     final int priorIndex = facts.getPriorIndex(key);
 
     Aggregator[] aggs;
-
+    AggregatorFactory[] metrics = getMetrics();
+    AtomicInteger numEntries = getNumEntries();
+    AtomicLong sizeInBytes = getBytesInMemory();
```
Comment (surekhasaharan) [inline processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java]: these can be declared `final` as they are not being changed in this method.

File: processing/src/test/java/org/apache/druid/segment/incremental/OnheapIncrementalIndexBenchmark.java [hunk 3]
```diff
@@ -187,7 +182,9 @@ protected AddToFactsResult addToFacts(
       final Integer priorIdex = getFacts().getPriorIndex(key);
 
       Aggregator[] aggs;
-
+      AggregatorFactory[] metrics = getMetrics();
+      AtomicInteger numEntries = getNumEntries();
+      AtomicLong sizeInBytes = getBytesInMemory();
```
Comment (surekhasaharan) [inline processing/src/test/java/org/apache/druid/segment/incremental/OnheapIncrementalIndexBenchmark.java]: can be final here too.

File: processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java [hunk 4]
```diff
@@ -98,6 +98,7 @@
 import java.util.stream.Stream;
 
 /**
+ *
```
[the remark this answers is not in the document]
Comment (justinborromeo) [reply processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java]: Sorry, autoformatter added that.  I'll remove

File: processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java (hunk 1 above)
Comment (justinborromeo) [reply processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java]: You're right.  I saw that they were being used in the index benchmarks but the benchmarks are in the same package.

**Pull request merged**